File: python_julia/01_read_process.py
# .\env_gdansk\Scripts\activate
import pandas as pd

# 1- read .dta datasets in /data
file_h = pd.read_stata("data/it14ih.dta")
file_p = pd.read_stata("data/it14ip.dta")

# 2 - subset variables
file_h.columns.values  # list of variables
file_h = file_h[["hid", "hilabour", "nhhmem", "hpopwgt"]]

file_p.columns.values  # list of variables
file_p = file_p[["hid", "pid", "pilabour", "sex", "age", "marital",
                 "disabled", "educlev", "lfs", "status1", "ind1_c", "occ1_c"]]

# 3- merge datasets
file = file_p.merge(file_h, on="hid", how="left")

# 4- Arrange the data
file = file[(file["age"] >= 30) & (file["age"] <= 60) & (file["pilabour"] > 0)]

# * print number of NAs in each variable
file.shape
file.isnull().sum()

# * delete rows with NAs
file = file.dropna()

# write file to csv
file.to_csv("clean_data/it14i.csv", index=False)

File: python_julia/jl/01_read_process.jl
using DataFrames, StatFiles

# Read data
current_dir = dirname(@__FILE__)
parent_dir = dirname(current_dir)

df_h = DataFrame(load(parent_dir * "/data/it14ih.dta"))
df_p = DataFrame(load(parent_dir * "/data/it14ip.dta"))

# Subset variables
df_h = df_h[:,["hid", "hilabour", "nhhmem", "hpopwgt"]]
df_p = df_p[:,["hid", "pid", "pilabour", "sex", "age", "marital", "disabled", 
    "educlev", "lfs", "status1", "ind1_c", "occ1_c"]]

# Merge df_h and df_p
df = leftjoin(df_h, df_p, on = :hid)

# Filter rows
df = df[(df[:,:age] .>= 30) .& (df[:,:age] .<= 60) .& (df[:,:pilabour] .> 0), :]

# * delete rows with NAs
df = dropmissing(df)